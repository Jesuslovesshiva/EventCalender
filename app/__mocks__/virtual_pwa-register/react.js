import {useState} from 'react'

const useRegisterSW = jest.fn(() => {
  const [needRefresh, setNeedRefresh] = useState(true)

  return {
    needRefresh: [needRefresh, setNeedRefresh],
    updateServiceWorker: () => {}
  }
})

// eslint-disable-next-line import/prefer-default-export
export {useRegisterSW}
