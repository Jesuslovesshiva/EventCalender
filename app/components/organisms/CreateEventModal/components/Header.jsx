import React from 'react'
import {FormattedMessage} from 'react-intl'

import messages from '@/components/organisms/CreateEventModal/messages'

const Header = () => (
  <span className="font-heading-large">
    <FormattedMessage {...messages.title} />
  </span>
)

export default React.memo(Header)
