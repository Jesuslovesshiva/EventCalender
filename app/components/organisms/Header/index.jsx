import {faker} from '@faker-js/faker'
import {Link} from '@tanstack/react-location'
import classNames from 'classnames'
import React from 'react'

import DnsIcon from '@/components/atoms/Icon/Dns'
import NotificationsIcon from '@/components/atoms/Icon/Notifications'
import PermContactCalendarIcon from '@/components/atoms/Icon/PermContactCalendar'
import SearchIcon from '@/components/atoms/Icon/Search'
import Navigation from '@/components/molecules/Navigation'
import TeamSelect from '@/components/molecules/TeamSelect'

import logo from './logo_dummy.svg'

const ICON_CLASSNAMES = 'text-2xl cursor-pointer'
const USER = 'JD'
const OPTIONS = [
  {isDisabled: false, label: 'Team A', value: 'teamA'},
  {isDisabled: false, label: 'Team B', value: 'teamB'}
]

const Header = () => (
  <div className="bg-white shadow flex justify-between z-50">
    <Link
      className="w-[5.625rem] flex items-center justify-center min-w-[50px]"
      to="/"
    >
      <img className="h-8" src={logo} alt="Logo" />
    </Link>
    <div className="flex-1 text-white relative">
      <div
        className="
          absolute top-0 bottom-0 right-[100%] w-[8px] pointer-events-none overflow-hidden
          before:absolute before:inset-0 before:translate-x-[8px]
          before:shadow-[0_0_8px_0_rgba(0,0,0,0.07),0_1px_1px_0_rgba(0,0,0,0.14)]
        "
      />
      <Navigation
        settings={
          <div
            className="
              flex justify-start lg:justify-end items-center justify-self-start flex-1 gap-3
              mt-0.5 mb-1.5 mx-[26px] order-2 lg:order-3
            "
          >
            <DnsIcon className={ICON_CLASSNAMES} />
            <PermContactCalendarIcon className={ICON_CLASSNAMES} />
            <div className="relative">
              <NotificationsIcon className={ICON_CLASSNAMES} />
              <div
                className="
                  bg-silver shadow h-3 w-3 text-primary text-xs
                  flex items-center justify-center
                  rounded-full absolute top-0 right-0 -mr-1 -mt-1
                "
              >
                {faker.datatype.number({max: 9, min: 1})}
              </div>
            </div>
            <div
              className="
                h-7 w-7 pb-px font-heading-small-caps
                flex justify-center items-center
                border rounded-full border-white
                select-none hover:cursor-pointer
              "
            >
              {USER}
            </div>
          </div>
        }
        cta={
          <div className="flex-1 mx-[26px] mt-0.5 w-full order-1">
            <div className="text-gray-dark flex items-center justify-start gap-3">
              <TeamSelect
                placeholder={OPTIONS[0].label}
                options={OPTIONS}
                value={OPTIONS[0]}
              />
              <SearchIcon
                className={classNames(ICON_CLASSNAMES, 'text-white')}
              />
            </div>
          </div>
        }
      />
    </div>
  </div>
)

export default React.memo(Header)
