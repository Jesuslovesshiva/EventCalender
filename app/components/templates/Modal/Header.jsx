import classNames from 'classnames'
import PropTypes from 'prop-types'

const ModalHeaderTemplate = ({children, className}) => (
  <div className={classNames('px-6 py-[18px]', className)}>{children}</div>
)

ModalHeaderTemplate.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string
}

ModalHeaderTemplate.defaultProps = {
  className: null
}

export default ModalHeaderTemplate
