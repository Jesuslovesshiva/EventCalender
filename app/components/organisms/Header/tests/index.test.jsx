import {fireEvent} from '@testing-library/react'

import render from '@/utils/test-utils/custom-render'

import Header from '../index'

jest.mock('@tanstack/react-location', () => ({
  ...jest.requireActual('@tanstack/react-location'),
  Link({children, ...props}) {
    return (
      <a {...props} href={props.to}>
        {children}
      </a>
    )
  }
}))
jest.mock('@/components/molecules/Navigation', () => ({cta, settings}) => (
  <div data-testid="navigation">
    {cta}
    <div>Navigation</div>
    {settings}
  </div>
))
jest.mock('@faker-js/faker', () => {
  const originalModule = jest.requireActual('@faker-js/faker')
  return {
    ...originalModule,
    faker: {
      ...originalModule.faker,
      datatype: {
        number: jest.fn(() => 8)
      }
    }
  }
})

const renderComponent = () => render(<Header />)

describe('Header', () => {
  it('should match the snapshot', () => {
    const {container} = renderComponent()
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should match the snapshot if the team-menu is open', () => {
    const {container, getByRole} = renderComponent()
    const combobox = getByRole('combobox')
    fireEvent.input(combobox, {target: {value: 'O'}})
    expect(container.firstChild).toMatchSnapshot()
  })
})
