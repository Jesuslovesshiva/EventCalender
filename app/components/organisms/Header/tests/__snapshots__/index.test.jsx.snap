// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match the snapshot 1`] = `
<div
  class="bg-white shadow flex justify-between z-50"
>
  <a
    class="w-[5.625rem] flex items-center justify-center min-w-[50px]"
    href="/"
    to="/"
  >
    <img
      alt="Logo"
      class="h-8"
      src="SVG_PATH"
    />
  </a>
  <div
    class="flex-1 text-white relative"
  >
    <div
      class=" absolute top-0 bottom-0 right-[100%] w-[8px] pointer-events-none overflow-hidden before:absolute before:inset-0 before:translate-x-[8px] before:shadow-[0_0_8px_0_rgba(0,0,0,0.07),0_1px_1px_0_rgba(0,0,0,0.14)] "
    />
    <div
      data-testid="navigation"
    >
      <div
        class="flex-1 mx-[26px] mt-0.5 w-full order-1"
      >
        <div
          class="text-gray-dark flex items-center justify-start gap-3"
        >
          <div
            class="text-base w-full specific:w-fit css-b62m3t-container"
          >
            <span
              class="css-1f43avz-a11yText-A11yText"
              id="react-select-2-live-region"
            />
            <span
              aria-atomic="false"
              aria-live="polite"
              aria-relevant="additions text"
              class="css-1f43avz-a11yText-A11yText"
            />
            <div
              class="!min-h-0 !rounded-1 !transition-shadow pl-3 gap-1.5 !bg-transparent !border-gray-medium hover:!border-surface hover:!bg-silver hover:!shadow-md !rounded-full !border-white !bg-transparent h-[27px] hover:!border-white hover:!bg-transparent hover:!shadow-md css-1s2u09g-control"
            >
              <div
                class="!py-0 !px-0 css-319lph-ValueContainer"
              >
                <div
                  class="!m-0 !p-0 !text-gray-dark !text-white !mb-px css-qc6sy-singleValue"
                >
                  U16
                </div>
                <div
                  class="!m-0 !p-0 !text-gray-dark !text-white !m-0 !p-0 css-6j8wv5-Input"
                  data-value=""
                >
                  <input
                    aria-autocomplete="list"
                    aria-expanded="false"
                    aria-haspopup="true"
                    autocapitalize="none"
                    autocomplete="off"
                    autocorrect="off"
                    class=""
                    id="react-select-2-input"
                    role="combobox"
                    spellcheck="false"
                    style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                    tabindex="0"
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                class=" css-1hb7zxy-IndicatorsContainer"
              >
                <div
                  aria-hidden="true"
                  class="text-2xl !px-3 !py-1 !text-gray-medium !text-white !pr-[7px] !pl-0 !py-0 css-tlfecz-indicatorContainer"
                >
                  <div
                    class="fill-current w-[1em] h-[1em] inline-block"
                    data-icon="arrow_drop_down"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="fill-current w-[1em] h-[1em] inline-block text-2xl cursor-pointer text-white"
            data-icon="search"
          />
        </div>
      </div>
      <div>
        Navigation
      </div>
      <div
        class=" flex justify-start lg:justify-end items-center justify-self-start flex-1 gap-3 mt-0.5 mb-1.5 mx-[26px] order-2 lg:order-3 "
      >
        <div
          class="fill-current w-[1em] h-[1em] inline-block text-2xl cursor-pointer"
          data-icon="dns"
        />
        <div
          class="fill-current w-[1em] h-[1em] inline-block text-2xl cursor-pointer"
          data-icon="perm_contact_calendar"
        />
        <div
          class="relative"
        >
          <div
            class="fill-current w-[1em] h-[1em] inline-block text-2xl cursor-pointer"
            data-icon="notifications"
          />
          <div
            class=" bg-silver shadow h-3 w-3 text-primary text-xs flex items-center justify-center rounded-full absolute top-0 right-0 -mr-1 -mt-1 "
          >
            8
          </div>
        </div>
        <div
          class=" h-7 w-7 pb-px font-heading-small-caps flex justify-center items-center border rounded-full border-white select-none hover:cursor-pointer "
        >
          JD
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Header should match the snapshot if the team-menu is open 1`] = `
<div
  class="bg-white shadow flex justify-between z-50"
>
  <a
    class="w-[5.625rem] flex items-center justify-center min-w-[50px]"
    href="/"
    to="/"
  >
    <img
      alt="Logo"
      class="h-8"
      src="SVG_PATH"
    />
  </a>
  <div
    class="flex-1 text-white relative"
  >
    <div
      class=" absolute top-0 bottom-0 right-[100%] w-[8px] pointer-events-none overflow-hidden before:absolute before:inset-0 before:translate-x-[8px] before:shadow-[0_0_8px_0_rgba(0,0,0,0.07),0_1px_1px_0_rgba(0,0,0,0.14)] "
    />
    <div
      data-testid="navigation"
    >
      <div
        class="flex-1 mx-[26px] mt-0.5 w-full order-1"
      >
        <div
          class="text-gray-dark flex items-center justify-start gap-3"
        >
          <div
            class="text-base w-full specific:w-fit css-b62m3t-container"
          >
            <span
              class="css-1f43avz-a11yText-A11yText"
              id="react-select-3-live-region"
            />
            <span
              aria-atomic="false"
              aria-live="polite"
              aria-relevant="additions text"
              class="css-1f43avz-a11yText-A11yText"
            />
            <div
              class="!min-h-0 !rounded-1 !transition-shadow pl-3 gap-1.5 !bg-transparent !border-gray-medium hover:!border-surface hover:!bg-silver hover:!shadow-md !rounded-b-none !rounded-full !border-white !bg-transparent h-[27px] !rounded-b-full hover:!border-white hover:!bg-transparent hover:!shadow-md css-1s2u09g-control"
            >
              <div
                class="!py-0 !px-0 css-319lph-ValueContainer"
              >
                <div
                  class="!m-0 !p-0 !text-gray-dark !text-white !m-0 !p-0 css-1pndypt-Input"
                  data-value="O"
                >
                  <input
                    aria-autocomplete="list"
                    aria-controls="react-select-3-listbox"
                    aria-expanded="true"
                    aria-haspopup="true"
                    aria-owns="react-select-3-listbox"
                    autocapitalize="none"
                    autocomplete="off"
                    autocorrect="off"
                    class=""
                    id="react-select-3-input"
                    role="combobox"
                    spellcheck="false"
                    style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                    tabindex="0"
                    type="text"
                    value="O"
                  />
                </div>
              </div>
              <div
                class=" css-1hb7zxy-IndicatorsContainer"
              >
                <div
                  aria-hidden="true"
                  class="text-2xl !px-3 !py-1 !text-gray-medium rotate-180 !text-white !pr-[7px] !pl-0 !py-0 rotate-180 !pr-0 !pl-[7px] !text-white css-tlfecz-indicatorContainer"
                >
                  <div
                    class="fill-current w-[1em] h-[1em] inline-block"
                    data-icon="arrow_drop_down"
                  />
                </div>
              </div>
            </div>
            <div
              class="!m-0 !rounded-t-none !shadow !rounded-t css-26l3qy-menu"
              id="react-select-3-listbox"
            >
              <div
                class="!p-0 !rounded-t css-4ljt47-MenuList"
              >
                <div
                  class=" css-gg45go-NoOptionsMessage"
                >
                  No options
                </div>
              </div>
            </div>
          </div>
          <div
            class="fill-current w-[1em] h-[1em] inline-block text-2xl cursor-pointer text-white"
            data-icon="search"
          />
        </div>
      </div>
      <div>
        Navigation
      </div>
      <div
        class=" flex justify-start lg:justify-end items-center justify-self-start flex-1 gap-3 mt-0.5 mb-1.5 mx-[26px] order-2 lg:order-3 "
      >
        <div
          class="fill-current w-[1em] h-[1em] inline-block text-2xl cursor-pointer"
          data-icon="dns"
        />
        <div
          class="fill-current w-[1em] h-[1em] inline-block text-2xl cursor-pointer"
          data-icon="perm_contact_calendar"
        />
        <div
          class="relative"
        >
          <div
            class="fill-current w-[1em] h-[1em] inline-block text-2xl cursor-pointer"
            data-icon="notifications"
          />
          <div
            class=" bg-silver shadow h-3 w-3 text-primary text-xs flex items-center justify-center rounded-full absolute top-0 right-0 -mr-1 -mt-1 "
          >
            8
          </div>
        </div>
        <div
          class=" h-7 w-7 pb-px font-heading-small-caps flex justify-center items-center border rounded-full border-white select-none hover:cursor-pointer "
        >
          JD
        </div>
      </div>
    </div>
  </div>
</div>
`;
